analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionQuerySimplification.cs

```diff
@@ -162,18 +163,40 @@ private static bool ParametersMatch(IMethodSymbol originalDefinition, IMethodSym
             return true;
         }
 
-        private static bool IsToCollectionCall(IMethodSymbol methodSymbol)
+        private static bool IsToCollectionCall(IMethodSymbol methodSymbol) =>
+            MethodNamesToCollection.Contains(methodSymbol.Name)
+            && (methodSymbol.IsExtensionOn(KnownType.System_Collections_Generic_IEnumerable_T)
+                || methodSymbol.ContainingType.ConstructedFrom.Is(KnownType.System_Collections_Generic_List_T));
+
+        private static string GetToCollectionCallsMessage(SonarSyntaxNodeReportingContext context, InvocationExpressionSyntax invocation, IMethodSymbol methodSymbol) =>
+            IsWithinEntityFrameworkContext(invocation, context.SemanticModel)
+                ? string.Format(MessageUseInstead, "'AsEnumerable'")
+                : string.Format(MessageDropFromMiddle, methodSymbol.Name);
+
+        private static bool IsWithinEntityFrameworkContext(InvocationExpressionSyntax node, SemanticModel model)
```

Comment:
Hint: This check as implemented has nothing to do with EF.

Reply:
But it would work for `IsInExpressionTree(invocation.Expression)` or `IsInExpressionTree((invocation.Expression as Invocation)?.Expression)`. The problem is that your check misses query comprehension syntax
```cs
_ = (from v in dbContext.MyEntities
     orderby v.Id
     select v).ToList().Where(SomeTest).ToList();
```